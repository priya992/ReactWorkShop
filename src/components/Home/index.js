import React from 'react';
import Card from '../Card';

const Home = ({ data, totalData }) => {
  const cards = data?.length>0 && data.map((value, index) => (
    <div  className="flex-row-item" key={index}>
      <Card value={value}/>
    </div>
  ))

  return <div className="flex-row-container">{cards}</div>
}

export default Home;
