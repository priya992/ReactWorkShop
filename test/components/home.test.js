import React from 'react';
import renderer from 'react-test-renderer';

import Home from './../../src/components/Home';
import { homeData } from './../__mock__';

test('Home component snapshot', () => {
  const component = renderer.create(
    <Home data={homeData.data} totalData={homeData.totalapps}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
