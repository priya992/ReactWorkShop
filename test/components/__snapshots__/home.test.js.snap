// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component 1`] = `
<div
  className="flex-row-container"
>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster1.jpg"
    />
    <div
      className="card-text"
    >
      The Birds
    </div>
  </div>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster2.jpg"
    />
    <div
      className="card-text"
    >
      Rear Window
    </div>
  </div>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster3.jpg"
    />
    <div
      className="card-text"
    >
      Family Pot
    </div>
  </div>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster2.jpg"
    />
    <div
      className="card-text"
    >
      Family Pot
    </div>
  </div>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster1.jpg"
    />
    <div
      className="card-text"
    >
      Rear Window
    </div>
  </div>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster3.jpg"
    />
    <div
      className="card-text"
    >
      The Birds
    </div>
  </div>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster3.jpg"
    />
    <div
      className="card-text"
    >
      Rear Window
    </div>
  </div>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster2.jpg"
    />
    <div
      className="card-text"
    >
      The Birds
    </div>
  </div>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster1.jpg"
    />
    <div
      className="card-text"
    >
      Family Pot
    </div>
  </div>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster1.jpg"
    />
    <div
      className="card-text"
    >
      The Birds
    </div>
  </div>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster1.jpg"
    />
    <div
      className="card-text"
    >
      The Birds
    </div>
  </div>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster2.jpg"
    />
    <div
      className="card-text"
    >
      Rear Window
    </div>
  </div>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster3.jpg"
    />
    <div
      className="card-text"
    >
      Family Pot
    </div>
  </div>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster2.jpg"
    />
    <div
      className="card-text"
    >
      Family Pot
    </div>
  </div>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster1.jpg"
    />
    <div
      className="card-text"
    >
      Rear Window
    </div>
  </div>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster3.jpg"
    />
    <div
      className="card-text"
    >
      The Birds
    </div>
  </div>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster3.jpg"
    />
    <div
      className="card-text"
    >
      Rear Window
    </div>
  </div>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster2.jpg"
    />
    <div
      className="card-text"
    >
      The Birds
    </div>
  </div>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster1.jpg"
    />
    <div
      className="card-text"
    >
      Family Pot
    </div>
  </div>
  <div
    className="flex-row-item"
  >
    <img
      alt="test"
      className="image-style"
      onError={[Function]}
      src="media/poster1.jpg"
    />
    <div
      className="card-text"
    >
      The Birds
    </div>
  </div>
</div>
`;
